mopidy/core/tracklist.py

```diff
@@ -381,7 +397,21 @@ def add(self, tracks=None, at_position=None, uris=None):
 
         tracks is None or validation.check_instances(tracks, Track)
         uris is None or validation.check_uris(uris)
-        validation.check_integer(at_position or 0)
+
+        if isinstance(at_position, (str, AddAtPosition)):
+            at_position = validation.check_enum(
+                at_position,
+                AddAtPosition,
+                msg="Expected an integer, or a member of {enum_name}: {enum_members}",
+            )
+        elif at_position is not None and not isinstance(at_position, int):
+            err_msg = "Expected an integer, or a member of {enum_name}: {enum_members}"
+            raise exceptions.ValidationError(
+                err_msg.format(
+                    enum_name=AddAtPosition.__name__,
+                    enum_members=", ".join(AddAtPosition.__members__.keys()),
+                )
+            )
```

Comment:
Or can we do this instead?
```suggestion
        if at_position is not None and not isinstance(at_position, int):
            at_position = validation.check_enum(
                at_position,
                AddAtPosition,
                msg="Expected an integer, or a member of {enum_name}: {enum_members}",
            )
```

Reply:
Thanks, this makes a lot more sense logically.